### PR TITLE
Fix admin config page error, use tests to cover the admin config and 500 error page

### PR DESCRIPTION
**Pull request by wxiaoguang**
The admin config page has been broken for many many times, a little refactoring would make this page panic.

So, add a test for it, and add another test to cover the 500 error page.

### CODE REVIEW

File: templates/admin/config.tmpl [hunk 1]
```diff
@@ -333,16 +333,6 @@
...
-				<dt>{{.locale.Tr "admin.config.git_enable_reflogs"}}</dt>
-				<dd>{{if .Git.Reflog.Enabled}}{{svg "octicon-check"}}{{else}}{{svg "octicon-x"}}{{end}}</dd>
-
-				{{if .Git.Reflog.Enabled}}
-					<dt>{{.locale.Tr "admin.config.git_reflog_expiry_time"}}</dt>
-					<dd>{{.locale.Tr "tool.days" .Git.Reflog.Expiration}}</dd>
-				{{end}}
-
-				<div class="ui divider"></div>
-
```
Comment (silverwind) [inline templates/admin/config.tmpl:345]: The git config parts are still shown, right? Care to show a screenshot?

Comment (wxiaoguang) [reply templates/admin/config.tmpl:345]: image

**Review by silverwind — APPROVED** [review]

File: routers/common/errpage.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func RenderPanicErrorPage(w http.ResponseWriter, req *http.Request, err any) {
 
 	defer func() {
 		if err := recover(); err != nil {
-			log.Error("Panic occurs again when rendering error page: %v", err)
+			log.Error("Panic occurs again when rendering error page: %v. Stack:\n%s", err, log.Stack(2))
```
Comment (delvh) [inline routers/common/errpage.go:29]: Wasn't that removed recently for some reason?

**Review by lunny — APPROVED** [review]

File: tests/integration/admin_config_test.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"code.gitea.io/gitea/tests"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAdminConfig(t *testing.T) {
+	defer tests.PrepareTestEnv(t)()
+
+	session := loginUser(t, "user1")
+	req := NewRequest(t, "GET", "/admin/config")
```
Comment (lunny) [inline tests/integration/admin_config_test.go:20]: Maybe we can have more links from `/admin` to test?

**Review by delvh — APPROVED** [review]

File: routers/common/errpage.go (hunk 2 above)
Comment (wxiaoguang) [reply routers/common/errpage.go:29]: Nope, not "removed", this line seems newly written, I forgot to add the stack here .... 😭

**Pull request merged**